orange/src/main/java/com/hartwig/hmftools/orange/algo/purple/PurpleVariantContextLoader.java

```diff
@@ -0,0 +1,219 @@
+                .alleleReadCount(tumorDepth.alleleReadCount())
+                .spliceRegion(variantImpact.CanonicalSpliceRegion)
+                .type(contextDecorator.type())
+                .gene(variantImpact.CanonicalGeneName)
+                .ref(contextDecorator.ref())
+                .alt(contextDecorator.alt())
+                .canonicalTranscript(variantImpact.CanonicalTranscript)
+                .canonicalEffect(variantImpact.CanonicalEffect)
+                .canonicalCodingEffect(variantImpact.CanonicalCodingEffect)
+                .canonicalHgvsCodingImpact(variantImpact.CanonicalHgvsCoding)
+                .canonicalHgvsProteinImpact(variantImpact.CanonicalHgvsProtein)
+                .worstCodingEffect(variantImpact.WorstCodingEffect)
+                .otherImpacts(otherImpacts)
+                .hotspot(contextDecorator.hotspot())
+                .reported(contextDecorator.reported())
+                .rnaDepth(rnaDepth)
+                .adjustedCopyNumber(contextDecorator.adjustedCopyNumber())
+                .adjustedVAF(contextDecorator.adjustedVaf())
+                .minorAlleleCopyNumber(contextDecorator.minorAlleleCopyNumber())
+                .variantCopyNumber(contextDecorator.variantCopyNumber())
+                .biallelic(contextDecorator.biallelic())
+                .genotypeStatus(contextDecorator.genotypeStatus(reference))
+                .repeatCount(contextDecorator.repeatCount())
+                .subclonalLikelihood(variantContext.getAttributeAsDouble(SUBCLONAL_LIKELIHOOD_FLAG, 0))
+                .localPhaseSets(localPhaseSets)
+                .build();
+    }
+
+    /**
+     * When VariantTranscriptImpacts are created from the VCF file, it sometimes (incorrectly) parses the square array brackets.
+     * These parsed brackets are then included in the impacts fields and here we remove them.
+     * TODO maybe fix this bug upstream in the VariantTranscriptImpact class?
+     */
+    @NotNull
+    private static VariantTranscriptImpact cleanVariantTranscriptImpactFields(VariantTranscriptImpact impact)
+    {
+        var cleanedGeneId = stripSquareBracketsAndWhiteSpace(impact.GeneId);
+        var cleanedGeneName = stripSquareBracketsAndWhiteSpace(impact.GeneName);
+        var cleanedTranscript = stripSquareBracketsAndWhiteSpace(impact.Transcript);
+        var cleanedHgvsCoding = stripSquareBracketsAndWhiteSpace(impact.HgvsCoding);
+        var cleanedHgvsProtein = stripSquareBracketsAndWhiteSpace(impact.HgvsProtein);
+        var cleanedEffects = stripSquareBracketsAndWhiteSpace(impact.Effects);
+        return new VariantTranscriptImpact(
+                cleanedGeneId,
+                cleanedGeneName,
+                cleanedTranscript,
+                cleanedEffects,
+                impact.SpliceRegion,
+                cleanedHgvsCoding,
+                cleanedHgvsProtein);
+    }
+
+    @Nullable
+    private static String stripSquareBracketsAndWhiteSpace(String s)
```

Comment:
Minor detail but would use `string` rather than `s`

Also, not entirely clear to me whether this logic is explicitly tested somewhere? I would at least test that an input VariantTranscriptImpact with brackets is handled correctly (though might have skipped over it, maybe `testCanReadVcfFile` tests it implicitly?).